dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -148,6 +155,14 @@ public static Result run(SolverContext ctx, ProverEnvironment prover, Refinement
         logger.debug("Final solver iteration:\n" +
                         " ===== Final Iteration: {} =====\n" +
                         "Native Solving/Proof time(ms): {}", iterationCount, curTime - lastTime);
+		
+        if(logger.isDebugEnabled()) {        	
+            String smtStatistics = "\n ===== SMT Statistics (after final iteration) ===== \n";
+    		for(String key : prover.getStatistics().keySet()) {
+    			smtStatistics += String.format("\t%s -> %s\n", key, prover.getStatistics().get(key));
+    		}
+    		logger.debug(smtStatistics);
+        }
```

Comment:
I think this should be at the very end of the verification (after the bound check) so it is aligned with the way the statistics of the other solving methods are produced.